This file is part of mingw-cross-env.
See doc/index.html for further information.

Patch taken from
http://code.google.com/p/libarchive/source/detail?r=3649

mingwrt 3.20 provides ssize_t and defines _SSIZE_T_. This patch
allows libarchive to use this ssize_t instead of defining its
own which causes build failure. Applies to libarchive 2.8.4.

diff --git a/libarchive/archive.h b/libarchive/archive.h
index 5ca28eb..f33a8ed 100644
--- a/libarchive/archive.h
+++ b/libarchive/archive.h
@@ -52,7 +52,7 @@
 /* These should match the types used in 'struct stat' */
 #if defined(_WIN32) && !defined(__CYGWIN__)
 #define	__LA_INT64_T	__int64
-# if defined(_SSIZE_T_DEFINED)
+# if defined(_SSIZE_T_DEFINED) || defined(_SSIZE_T_)
 #  define	__LA_SSIZE_T	ssize_t
 # elif defined(_WIN64)
 #  define	__LA_SSIZE_T	__int64
-- 
1.7.6

